pkg/autodiscovery/scheduler/loader.go

```diff
@@ -0,0 +1,33 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package scheduler
+
+import (
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+)
+
+// Catalog holds available schedulers
+type Catalog map[string]Scheduler
+
+// DefaultCatalog holds every compiled-in diagnosis
+var DefaultCatalog = make(Catalog)
```

Comment:
We're using the `Catalog` and `DefaultCatalog` in different ways than other packages: in all other packages, the `DefaultCatalog` is statically filled, and selecting useful entities is done during `Start`.

Here, the `DefaultCatalog` is filled at runtime depending on the agent configuration. Do avoid future head scratching, let's document this difference. We might also want to rename these variables to something else to further avoid confusion? Maybe that's an `ActiveSchedulers` variable we have here?